src/views/Components/Controls/Input.js

```diff
@@ -0,0 +1,137 @@
+import React from 'react';
+import styled from 'styled-components';
+import PropTypes from 'prop-types';
+
+const selectWrapperConditionStyle = (condition, successColor, dangerColor) => {
+  if (typeof (condition) === 'boolean') {
+    if (condition) {
+      return `
+      border-color: ${successColor};
+      box-shadow: 0 5px 11.5px rgba(23,198,113,.1);
+      `;
+    }
+
+    return `
+      border-color: ${dangerColor};
+      box-shadow: 0 5px 11.5px rgba(196,24,60,.1);
+      `;
+  }
+};
+
+const messageWrapperConditionStyle = (condition, successColor, dangerColor) => {
+  if (typeof (condition) === 'boolean') {
+    if (condition) {
+      return `
+        color: ${successColor};
+      `;
+    }
+
+    return `
+      color: ${dangerColor};
+      `;
+  }
+};
+
+const Wrapper = styled.div`
+  min-width: 7.5rem;
+  width: 100%;
+`;
+
+const IconWrapper = styled.div`
+  position: relative;
+  display: flex;
+  flex-wrap: nowrap;
+  align-items: stretch;
+  width: 100%;
+  min-width: 7.5rem;  
+`;
+
+const InputWrapper = styled.input`
+  padding: .4375rem .75rem;
+  flex: 1 1 auto;
+  font-size: .8125rem;
+  line-height: 1.5;
+  color: ${props => props.theme.colors.text};
+  background-color: ${props => props.theme.colors.white};
+  border: 1px solid ${props => props.theme.colors.border};
+  font-weight: 300;
+  border-radius: .25rem;
+  box-shadow: none;
+  transition: box-shadow 250ms cubic-bezier(.27,.01,.38,1.06), border 250ms cubic-bezier(.27,.01,.38,1.06);
+
+  &:hover {
+    border-color: ${props => props.theme.colors.hoverColors.border};
+  }
+
+  &:focus {
+    color: ${props => props.theme.colors.text};
+    background-color: ${props => props.theme.colors.white};
+    border-color: ${props => props.theme.colors.primary};
+    box-shadow: 0 0.313rem 0.719rem rgba(0,123,255,.1), 0 0.156rem 0.125rem rgba(0,0,0,.06);
+  }
+
+  ${({ condition, theme }) => selectWrapperConditionStyle(condition, theme.colors.success, theme.colors.danger)};
+
+  ${({ icon }) =>
+    icon && `
+    border-top-left-radius: 0;
+    border-bottom-left-radius: 0;
+    `
+};
+`;
+
+const MessageWrapper = styled.div`
+  display: block;
+  margin-top: .25rem;
+  font-size: 80%;
+
+  ${({ condition, theme }) => messageWrapperConditionStyle(condition, theme.colors.success, theme.colors.danger)};
```

Comment:
Please rewrite it.
messageWrapperConditionStyle should receive props and take all the data from there. It should be used like this. 
```
color: `${messageWrapperColor}`
```